lib/Sema/DerivedConformanceDifferentiable.cpp

```diff
@@ -167,7 +165,7 @@ bool DerivedConformance::canDeriveDifferentiable(NominalTypeDecl *nominal,
   };
 
   auto invalidTangentDecls = llvm::partition(tangentDecls, [&](ValueDecl *v) {
-    return isValidAssocTypeCandidate(v, /*checkAdditiveArithmetic*/ true);
```

Comment:
I wonder why this logic was changed? (checking whether `TangentVector` candidates conform to `AdditiveArithmetic`)

Reply:
The logic is redundant. This is the only caller of `isValidAssocTypeCandidate` and thus `checkAdditiveArithmetic` is never `false`.

Reply:
Gotcha, thanks.

Reply:
After offline discussion, I noticed that I missed another caller of `isValidAssocTypeCandidate`. This change is now reverted.